gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/dbViews/views.json

```diff
@@ -12,37 +12,40 @@
                 "map": "function(doc) {if (doc.type === 'gpiiToken') emit(doc.gpiiToken, {'_id': doc.userId})}"
             },
             "findClientByOauth2ClientId": {
-                "map": "function(doc) {if (doc.type === 'client') emit(doc.oauth2ClientId, doc)}"
+                "map": "function(doc) {if (doc.type === 'gpiiAppInstallationClient' || doc.type === 'privilegedPrefsCreatorClient' || doc.type === 'webPrefsConsumerClient') emit(doc.oauth2ClientId, doc)}"
             },
-            "findAllClients": {
-                "map": "function(doc) {if (doc.type === 'client') emit(doc._id, doc)}"
+            "findClientBySolutionId": {
```

Comment:
I hope @waharnum 's solution to this will find its way into the architecture soon :)